op/main.py

```diff
@@ -44,46 +45,50 @@ def _handle_event_link(charm_dir, bound_event):
         # (otherwise, we would never have been called).
         return
 
-    event_hook_path = charm_dir / 'hooks' / bound_event.event_kind.replace('_', '-')
+    if issubclass(bound_event.event_type, op.charm.HookEvent):
+        event_dir = charm_dir / 'hooks'
+    else:
+        raise RuntimeError(f'unable to create a symlink: unsupported event type {bound_event.event_type}')
+    # TODO: Handle function/action events here.
+
+    if not event_dir.exists():
+        raise RuntimeError(f'unable to create event symlink: {event_dir} directory does not exist')
+
+    event_path = event_dir / bound_event.event_kind.replace('_', '-')
     create_link = True
     # Remove incorrect symlinks or files.
-    if event_hook_path.exists():
-        # Non-symlink entries and the ones not pointing to "install"
-        # need to be removed.
-        if not event_hook_path.is_symlink():
-            debugf(f'Hook entry at {event_hook_path} is not a symlink:'
-                   ' attempting to remove it.')
+    if event_path.exists():
+        # Non-symlink entries and the ones not pointing to the charm code file need to be removed.
+        if not event_path.is_symlink():
+            debugf(f'Event entry at {event_path} is not a symlink: attempting to remove it.')
             # May raise IsADirectoryError, e.g. in case it is a directory which
             # is unexpected and left to the developer or operator to handle.
-            event_hook_path.unlink()
-        elif os.readlink(event_hook_path) != 'install':
-            debugf(f'Removing entry {event_hook_path} as it does not point'
-                   ' to "install"')
-            event_hook_path.unlink()
+            event_path.unlink()
+        elif os.readlink(event_path) != CHARM_CODE_FILE:
+            debugf(f'Removing entry {event_path} as it does not point to {CHARM_CODE_FILE}')
+            event_path.unlink()
         else:
             create_link = False
 
     if create_link:
-        debugf(f'Creating a new relative symlink at {event_hook_path}'
-               f' to pointing to "install" located in {charm_dir}/hooks')
-        event_hook_path.symlink_to('install')
+        debugf(f'Creating a new relative symlink at {event_path} to pointing to {CHARM_CODE_FILE}')
+        event_path.symlink_to(CHARM_CODE_FILE)
```

Comment:
As mentioned on Telegram, my preference is to drop the hard-coded `CHARM_CODE_FILE` value in favor of `os.readlink(sys.argv[0])`, which trusts the charm author to have gotten the first link correct and just reuses that link target for all other symlinks.

Reply:
Thanks for the suggestion, see https://github.com/canonical/operator/pull/63/files#r352704416